### PR TITLE
Inline validation in signup and login

### PR DESCRIPTION
**Pull request by AlfiyaSiddique**
## Description
Added inline validation in signup and login forms.

## Sample Output Video:

https://github.com/narayan954/dummygram/assets/86224794/2be2798b-5338-4b57-8645-b98d53b0ac99

### CODE REVIEW

**Review by narayan954 — CHANGES_REQUESTED** [review]
I doubt we need to validate login form for passwords, maybe I'm incorrect with that.You can tell me.
image
Also, there's one critical problem. You're validating and showing errors but not restricting any user from login/signup. we can still signup/login along with errors. Please fix that

File: src/pages/Signup/index.jsx [hunk 1]
```diff
@@ -243,25 +253,50 @@ const SignupScreen = () => {
...
             placeholder=" Email"
+            name="email"
             value={email}
-            onChange={(e) => setEmail(e.target.value)}
+            onChange={(e) => {
+              setEmail(e.target.value);
+              handleError(e.target.name, e.target.value);
+            }}
+            className={
+              error.emailError? "username-not-available": null
```
[the remark this answers is not in the document]
Comment (AlfiyaSiddique) [reply src/pages/Signup/index.jsx:283]: I used this class to make the borders red when an error exists, so I guess to make it readable and generalize it we can make rename the class as error-border, what do you say??

**Comment by AlfiyaSiddique** [comment]
> I doubt we need to validate login form for passwords, maybe I'm incorrect with that.You can tell me. image Also, there's one critical problem. You're validating and showing errors but not restricting any user from login/signup. we can still signup/login along with errors. Please fix that

About the login password, If the validation exists it will provide immediate feedback to the user to type the correct password in case they typed incorrect password validation-wise,  Also if you think there's no need for that then let me know I'll do as you say.

And about restricting form submission ,  I'll fix it as soon as possible and make  a commit.

**Comment by narayan954** [comment]
> > I doubt we need to validate login form for passwords, maybe I'm incorrect with that.You can tell me. image Also, there's one critical problem. You're validating and showing errors but not restricting any user from login/signup. we can still signup/login along with errors. Please fix that
> 
> About the login password, If the validation exists it will provide immediate feedback to the user to type the correct password in case they typed incorrect password validation-wise, Also if you think there's no need for that then let me know I'll do as you say.
> 
> And about restricting form submission , I'll fix it as soon as possible and make a commit.

you may remove the password validation at login, because old users won't be able to login.

**Comment by AlfiyaSiddique** [comment]
> > > I doubt we need to validate login form for passwords, maybe I'm incorrect with that.You can tell me. image Also, there's one critical problem. You're validating and showing errors but not restricting any user from login/signup. we can still signup/login along with errors. Please fix that
> > 
> > 
> > About the login password, If the validation exists it will provide immediate feedback to the user to type the correct password in case they typed incorrect password validation-wise, Also if you think there's no need for that then let me know I'll do as you say.
> > And about restricting form submission , I'll fix it as soon as possible and make a commit.
> 
> you may remove the password validation at login, because old users won't be able to login.

Okay, will do it soon. 👍🏻

**Comment by AlfiyaSiddique** [comment]
Hello @narayan954 

Implemented the changes you asked for, my new commits contain the following changes:
1. Rename the class username-not-allowed to error-border.
2. Implemented the code for preventing the form from getting submitted if errors exist.
3. Remove the password validation of the login form.

**Comment by narayan954** [comment]
> Hello @narayan954
> 
> Implemented the changes you asked for. My new commits contain the following changes:
> 
> 1. Rename the class **username-not-available** to **error-border**.
> 2. Implemented the code for preventing the form from getting submitted if errors exist.
> 3. Remove the password validation of the login form.

Thanks for fixing them, reviewing

**Review by narayan954 — APPROVED** [review]
looks good to me!

File: src/pages/Signup/index.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (narayan954) [reply src/pages/Signup/index.jsx]: yes that'd be great

**Pull request merged**